### ISSUE TITLE
make all not finding install media even after running installinstallmacos.py

### ISSUE THREAD

**Issue by EasyG0ing1** [AlexanderWillner/runMacOSinVirtualBox#127]
When I execute make all (on my MacBook Pro running 11.2.1 in iTerm2) it comes back with this:

`ERROR: No macOS installer found at /Applications. Download the installer first (e.g. via 'installinstallmacos.py') - press enter in the terminal when done...
`
So I open a new tab in iTerm and change into the same folder and run installinstallmacos.py - choosing Big Sur as the OS I want it to download. It downloads and does everything just fine and it leaves a .DMG file in the same folder where I executed the python script.  That filename is **Install_macOS_11.3-20E5210c.dmg**

So when that script is done, I go back to the other tab and hit enter just like the instructions say and it then comes back with this:

```
DEBUG: line 98 - command 'exit 6' exited with status: 6.
DEBUG: In ::runChecks::main::main called at line 410.
DEBUG: From function ::runChecks::main::main (line 410).
ERROR: Look at /Users/michael/Library/Logs/runMacOSVirtualbox.log for details (or use Console.app). Press enter in the terminal when done...
```

Then when I hit enter, it gives me this and kicks me back out to the shell prompt:

`make: *** [all] Error 6`

I did try moving the DMG file into /Applications and ran make all but it gave the same errors. I didn't expect that to work of course because I realize that the script is looking for the standard .app file that gets downloaded from the App Store when you download a mac operating system.

But the installinstallmacos.py script obviously doesn't download a .app package ...

So what is the answer here? How do I tell **make all** to use the DMG file that was downloaded from the script?

Thank you,

Mike Sims

**Comment by Khrystarlite** [comment]
installinstallmacos.py produces a dmg file.
If you double click on/open the dmg file, the app is inside.
I recommend this be added to the README

**Comment by swiftavl** [comment]
https://github.com/AlexanderWillner/runMacOSinVirtualBox/issues/127

**TLDR: I think the following should work, otherwise, read the whole thing below.**
> ./installinstallmacos.py

open the dmg file and move the installer app file to the applications folder

> ./runMacOSVirtualbox.sh installer
> make all

**The whole thing:**
I am installing this so I can read some old HFS CDs. What a pain and waste of time. HFS Utilities works, but could not figure out how to escape '&' and get to directories like that.

**I'm on a MacBook Pro 16" with 16GB ram running macOS Catalina 10.15.7. I installed Virtualbox using brew.
>brew install virtualbox
>brew install virtualbox-extension-pack**

I had the same issue stated here. I think it might have something to do with the make file having an issue creating the iso.cdr file. Don't worry, the included shell script can. Here is approximately what I did and it worked. Some steps are probably not needed, but I did not go back and test which steps were not needed.

> ./installinstallmacos.py

I chose option 9 which was Install_macOS_10.14.6-18G103.dmg
Moved the dmg to the applications folder, opened the dmg file and copied the 'Install macOS Mojave.app' to the Applications folder.

I had already run make all so many times and gotten errors, I decided to try something different. I deleted the macOS-VM.dmg from the VirtualBox VMs folder again.

I ran the other script.
> ./runMacOSVirtualbox.sh

it comes back with a message:
> Possible commands: clean, stash, all, check, installer, patch, vm, run, stop, wait, eject, add

Then I ran
> ./runMacOSVirtualbox.sh check
> ./runMacOSVirtualbox.sh clean
> ./runMacOSVirtualbox.sh check
> ./runMacOSVirtualbox.sh installer

It asked for sudo password and everything worked to make the macOS-VM.iso.cdr

And then I tried make again.
> make all

I will not speculate on where the issue is, but here are a couple error messages that came up:
> DEBUG: line 98 - command 'exit 6' exited with status: 6.
> DEBUG: In ::runChecks::main::main called at line 410.
> DEBUG: From function ::runChecks::main::main (line 410).
and
> DEBUG: line 420 - command '[ "$#" -ne 0 ]' exited with status: 1.
> DEBUG: In :: called at line 0.
> DEBUG: From function :: (line 0).

After that, things came up in VirtualBox. It hung at 'about 1 minute' left in the macOS install. Fix was rebooting the virtual machine. After linking the USB to SATA adapter in VirtualBox and setting up File Sharing, I can now read HFS formatted CDs and write them to an external SSD.

Thank you for this project and hope the above helps somebody.
It could be I read the documentation wrong.

**Comment by EasyG0ing1** [comment]
I ended up finding a script called "macos-guest-virtualbox.sh" and it worked well... fairly complex and involved script with some strict requirements like being on the latest version of BASH etc. but once you meet the pre-lims it was flawless.
